coreneuron/io/reports/report_configuration_parser.cpp

```diff
@@ -59,62 +61,49 @@ enum class TargetType
 /*
  * Split filter string ("mech.var_name") into mech_id and var_name
  */
-void parse_filter_string(char* filter, ReportConfiguration& config) {
-    char* token = strtok(filter, ".");
-    if (!token) {
-        std::cerr << "Error : Invalid report variable, should be mch_name.var_name" << std::endl;
-        nrn_abort(1);
-    }
-    strcpy(config.mech_name, token);
-    token = strtok(nullptr, "\n");
-    if (!token) {
-        std::cerr << "Error : Invalid report variable, should be mch_name.var_name" << std::endl;
-        nrn_abort(1);
-    }
-    strcpy(config.var_name, token);
+void parse_filter_string(const std::string &filter, ReportConfiguration &config) {
+    std::istringstream iss(filter);
+    std::string token;
+    std::getline(iss, config.mech_name, '.');
```

Comment:
Just curious, will this fail if the `.` is not found like the original code?

Reply:
I thought about that, but if it is not x.y format, neurodamus fails already